ios-target-app/Home/ViewModels/TargetViewModel.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  TargetViewModel.swift
+//  ios-target-app
+//
+//  Created by Raul Piñeres Carrera on 1/11/22.
+//
+
+import Foundation
+
+enum TargetViewModelState {
+  case created
+  case network(state: NetworkState)
+}
+
+final class TargetViewModel {
+  
+  private let targetServices: TargetServices
+  
+  init(targetServices: TargetServices = TargetServices()) {
+    self.targetServices = targetServices
+  }
+
+  @Published private var state: TargetViewModelState = .network(state: .idle)
+  var statePublisher: Published<TargetViewModelState>.Publisher { $state }
```

Comment:
why do you need both publishers?
you could have something like:
```suggestion
  @Published private(set) var state: TargetViewModelState = .network(state: .idle)
```